pkg/archive/archive_test.go

```diff
@@ -51,55 +53,51 @@ func defaultCopyWithTar(src, dst string) error {
 }
 
 func TestIsArchivePathDir(t *testing.T) {
-	cmd := exec.Command("sh", "-c", "mkdir -p /tmp/archivedir")
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		t.Fatalf("Fail to create an archive file for test : %s.", output)
-	}
-	if IsArchivePath(tmp + "archivedir") {
+	if IsArchivePath(t.TempDir()) {
 		t.Fatalf("Incorrectly recognised directory as an archive")
 	}
 }
 
 func TestIsArchivePathInvalidFile(t *testing.T) {
-	cmd := exec.Command("sh", "-c", "dd if=/dev/zero bs=1024 count=1 of=/tmp/archive && gzip --stdout /tmp/archive > /tmp/archive.gz")
+	tmpDir := t.TempDir()
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("dd if=/dev/zero bs=1024 count=1 of=%[1]s/archive && gzip --stdout %[1]s/archive > %[1]s/archive.gz", toUnix(t, tmpDir)))
 	output, err := cmd.CombinedOutput()
 	if err != nil {
 		t.Fatalf("Fail to create an archive file for test : %s.", output)
 	}
-	if IsArchivePath(tmp + "archive") {
+	if IsArchivePath(filepath.Join(tmpDir, "archive")) {
```

Comment:
These were done in https://github.com/moby/go-archive/commit/e52d6b0d7d9be464d202708a0eee32655fe861e1